subprojects/dependency-management/src/main/java/org/gradle/internal/resolve/result/DefaultBuildableComponentSelectionResult.java

```diff
@@ -1,96 +0,0 @@
-/*
- * Copyright 2015 the original author or authors.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.gradle.internal.resolve.result;
-
-import org.gradle.api.artifacts.component.ModuleComponentIdentifier;
-import org.gradle.internal.resolve.ModuleVersionResolveException;
-
-import javax.annotation.Nullable;
-import java.util.LinkedHashSet;
-import java.util.Set;
-
-public class DefaultBuildableComponentSelectionResult implements BuildableComponentSelectionResult {
```

Comment:
It seems that this was intended to encapsulate some state instead of putting it all in one class. Was this never used anywhere?

Reply:
It was used as a convenience to store intermediate selection result, but in practice not all state was necessary. So I merged it where it was used, making the algorithm cleaner while reducing the number of objects we create. See c0dc85f